src/ship.c

```diff
@@ -741,6 +811,9 @@ static int ship_parse( Ship *temp, xmlNodePtr parent )
          /* Load the graphics. */
          ship_loadGFX( temp, buf, sx, sy, engine );
 
+         /* Load the polygon. */
+         ship_loadPLG( temp, buf );
```

Comment:
Should warn if no polygon is available and give a hint or something on how to generate it.

Reply:
If we can generate polygons completely reliably, we might want to just completely remove the code for sprite collisions, and error if there isn't a polygon.

Reply:
> If we can generate polygons completely reliably, we might want to just completely remove the code for sprite collisions

I think it's quite risky for now... I would say that we can generate polygons semi-reliably. Indeed, afaik, there is no algo that works in all cases to generate non-convex polygons.
However, thanks to the fact that in our case, the smaller size is a pixel, I managed to write an algo that always works, at the price of extreme refinement of the polygon (with lines of 1 pixel long). Experience shows that this extreme refinement is not needed more than on 1 or 2 sprites per ship for the most complex shapes, which is enough to ensure good performance of the algo.
But I cannot guarantee it's not possible to design a ship that would request such refined polygons that it would make the algo unusable.

I'd prefer adding a warning.